### PR TITLE
feat: add app context, update packages, get lang from querystring

### LINKED ISSUE #87: Get language from query string and context

### PR DESCRIPTION
**Pull request by pyphilia**
I added the app context, it's how we use it in the new graasp. So it allows language and context (teacher/student mode). 
I've added an additional logic to use the query string for lang.

close #87

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -11,8 +11,16 @@
...
+  "parser": "@babel/eslint-parser",
+  "parserOptions": {
+    "requireConfigFile": false,
+    "babelOptions": {
+      "presets": ["@babel/preset-react"]
+    }
+  },
   "rules": {
+    // TO REMOVE
+    "class-methods-use-this": "off",
```
Comment (spaenleh) [inline .eslintrc:23]: Remove now or later ?

File: src/components/App.js [hunk 2]
```diff
@@ -1,89 +1,57 @@
...
-  }
-
-  componentDidUpdate({ lang: prevLang }) {
-    const { lang } = this.props;
-    // handle a change of language
-    if (lang !== prevLang) {
-      this.handleChangeLang(lang);
-    }
-  }
+export const App = ({ i18n }) => {
```
Comment (spaenleh) [inline src/components/App.js:11]: You can get the current i18n instance with the `useTransations` hook if that can be useful ?

File: src/components/App.js [hunk 3]
```diff
@@ -1,89 +1,57 @@
...
+App.propTypes = {
+  i18n: PropTypes.shape({
+    defaultNS: PropTypes.string,
+    changeLanguage: PropTypes.func,
+    language: PropTypes.string,
+  }).isRequired,
+};
 
-export default withTranslation()(ConnectedApp);
+export default withTranslation()(App);
```
Comment (spaenleh) [inline src/components/App.js:57]: okay I see you kept the `withTanslation` HOC.

**Review by spaenleh — APPROVED** [review]
Looks good to me, we just need to test to ensure everything works as expected :)

File: .eslintrc (hunk 1 above)
Comment (pyphilia) [reply .eslintrc:23]: later..

**Comment by pyphilia** [comment]
@hagoptaminian in the end I won't merge it now, since we don't urgently need languages. You can freely create your PR and merge before this one. I'll tackle this one (one day) to fully include the new Graasp mechanisms.

**Pull request closed without merging**